prog/vm/nexus.rb

```diff
@@ -263,6 +263,9 @@ def before_run
       nap 30
     end
 
+    queued_vm_count = Vm.join(:strand, id: :id).where(:location => vm.location, Sequel[:strand][:label] => "start").count
```

Comment:
Is there a more elegant way to factor this with the `queued_vms` expression above?  It is largely repeated.

Also, do we we need to move stuff to external symbols or make the label smaller so that your test can be a bit less brutal?

Reply:
I made both `queued_vms` and `queued_vm_count` to use same Sequel construct. 

However moving stuff to another label seems more troublesome because we want to rollback host allocation if ip assignment fails as well, which requires all ip selection relates stuff to be in the same label with the host selection. Instead, I refactored the tests a bit by moving some stuff to `before` block